docs/deprecation.md

```diff
@@ -0,0 +1,37 @@
+# Deprecation Policy
+
+In order for any software to be reliable enough for production use, it needs to have a stable API. This means that
+users would have a high confidence that the way they use Kubevirt would not become obsolete, or at least would give
+the proper time to prepare if it does.
+
+This document aims to specify a process for several aspects of deprecation revolving Kubevirt. This is intended to
+be a "live document" which updates and refines itself over time.
+
+## Process
+
+### Feature Gates
+#### Introduction
+A feature gate should be introduced for every new feature, capability, configuration or any other change that
+is not yet stable and can jeopardize the cluster.
+
+A feature gate serves as:
+* A way to warn the user about a possible danger that could be caused by an unstable logic.
+* A way for the users to explicitly state that they would like to take the risk and enable the feature.
+
+Over time the new features would stabilize and graduate their API versions (i.e. from beta, to alpha to stable). When
+that happens, and therefore there risk of jeopardizing the user decreases significantly, the feature should be enabled
+by default and the feature gate should be deprecated (see section below).
+
+#### Deprecation
+When a feature stabilizes and there is no need to guard it by a feature gate, the feature gate should be deprecated.
+
+In order to deprecate a feature gate the following conditions must be met:
+* All the features that used to be guarded by it need to be enabled by default.
```

Comment:
yea. today our feature gates are stored in an array ([]string). Feature gate is getting enabled if its key is present in that array.
To enable a feature gate by default would mean adding it to the kubevirt CR by default...

Reply:
What I meant by that is that feature gates would turn into a no-op, meaning no features would be guarded by the feature gate. But you're right, this isn't written in a clear way.

I've added this sub-bullet to clarify:
```
* The above point means that all of the features that were guarded by this feature gate
  would be enabled without the need to enable this feature gate. In other words, the
  feature gate needs to turn into a no-op.
```

WDYT?

Reply:
In the code, i imagine we'd just have a list of old feature gates that's only used to warn people during validation. Part of the path to moving a feature gate to this list of old gates would be to completely remove usage of the feature gate internally and ensure it's always on.

LiveMigration would be a good candidate to start with here :)

Reply:
We can't control what is in the Kubevirt CR so I would agree with the warning and no-op.